### PR TITLE
WARN: Avoid FutureWarnings in tests

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Would like to backport, otherwise we have to deal with this on the 1.5.x branch

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Sure fine to backport. Thanks @phofl

File: pandas/tests/frame/methods/test_update.py [hunk 1]
```diff
@@ -136,7 +136,8 @@ def test_update_from_non_df(self):
     def test_update_datetime_tz(self):
         # GH 25807
         result = DataFrame([pd.Timestamp("2019", tz="UTC")])
-        result.update(result)
+        with tm.assert_produces_warning(FutureWarning):
```
Comment (jbrockmendel) [inline pandas/tests/frame/methods/test_update.py:139]: i think this is wrong.  in DataFrame.update we use `self.loc[:, col] = ...` in a way that produces a warning that we probably don't want.  so we need to change what we're doing there, not catch a warning here

Comment (phofl) [reply pandas/tests/frame/methods/test_update.py:139]: Yeah, agreed. Forgot to post here. Hadn't had time to take a closer look yet though. We probably have a couple more of those cases

Comment (phofl) [reply pandas/tests/frame/methods/test_update.py:139]: Opened #48468 to revert the update warning

**Pull request merged**